src/components/common/forms/SignInForm.tsx

```diff
@@ -1,31 +1,63 @@
 import React from "react";
 import { useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+
 import { Button, Input } from "@components/common";
+import { signInRequest } from "@/services";
+import { urlFormat, setLocalStorage } from "@/utils";
+import { Pages } from "@/pages";
+
 import styles from "./Forms.module.css";
 
 const SignInForm = () => {
-  const { register, handleSubmit } = useForm();
-  const onSubmit = (data: object) => {
-    //logic of submission
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setError,
+    clearErrors,
+  } = useForm();
+  const navigate = useNavigate();
+
+  const onSubmit = async (data: object) => {
+    const response = await signInRequest({
+      user: data,
+    });
+    if (typeof response === "object") {
+      setLocalStorage("user", response);
+      navigate(urlFormat(Pages.Main));
+    } else {
+      setError("serverError", { type: "custom", message: response });
+    }
```

Comment:
I'm not sure to handle success/error based on response type, maybe you could add a try / catch in `onSubmit` or maybe some status inside response to handle if the request succeed ?

Reply:
Perfect with try/catch looks way better 

```suggestion
   const onSubmit = async (data: object) => {
    try {
      const response = await signInRequest({
        user: data,
      });
      setLocalStorage("user", response);
      navigate(urlFormat(Pages.Main));
    } catch (error) {
      setError("serverError", {
        type: "custom",
        message: (error as Error).message,
      });
    }
  };
```